compiler/rustc_attr_parsing/src/parser.rs

```diff
@@ -87,6 +87,14 @@ impl<'a> PathParser<'a> {
     pub fn word_is(&self, sym: Symbol) -> bool {
         self.word().map(|i| i.name == sym).unwrap_or(false)
     }
+
+    /// Checks whether the first segments match the givens.
+    ///
+    /// Unlike [`segments_is`](Self::segments_is),
+    /// `self` may contain more segments than the number matched  against.
+    pub fn starts_with(&self, segments: &[Symbol]) -> bool {
+        self.segments().zip(segments).all(|(a, b)| a.name == *b)
```

Comment:
should probably assert that `segments.len() < self.segments().len()`

Reply:
I think I stopped using it again, probably can be removed again?

Reply:
or we keep it to prevent the next person from making the mistake :thinking: